### PR TITLE
Add services reset password

### PR DESCRIPTION
**Pull request by CamiloArias12**
# Services reset password

## Description
Se agregaron los servicos para restablecer la contrasena del usuario ,en el cual se envia un correo al cliente con un token que permita identificar al usuario
### Summary
Se agregaron los servicios :
1.Enviar correo con token
2.Validar token
3.Restablecer contrasena

## How Has This Been Tested?
Se testeo directamente en la web

## Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: internal/service/client/client.go [hunk 1]
```diff
@@ -115,21 +117,82 @@ func (s *serviceClient) Login(req *dto.LoginClientReq) (*dto.LoginClientRes, err
...
+	}
+
+	token, err := security.JWTSign(client, cfg.Env.JWTSecretPassword)
+	if err != nil {
+		return false, err
+	}
+
+	if err := mailing.SendResetPassword(&dto_mailing.ResetPasswordMailingReq{
+		Email:   res.Email,
+		Subject: "Owlbytech Restablecer contrasena",
```
Comment (Camilo-Camargo) [inline internal/service/client/client.go:157]:
```suggestion
		Subject: "Restablecer contraseña",
```

File: internal/service/client/client.go [hunk 2]
```diff
@@ -115,21 +117,82 @@ func (s *serviceClient) Login(req *dto.LoginClientReq) (*dto.LoginClientRes, err
...
+
+	return true, nil
+}
+
+func (s *serviceClient) ResetPasswordToken(req *dto.ResetPasswordTokenReq) (bool, error) {
+	_, payload, err := security.JWTGetPayload(req.Token, cfg.Env.JWTSecretPassword)
+	if err != nil {
+		return false, err
+	}
+	client, err := s.Get(dto.GetClientReq{Id: int32(payload.ID)})
```
Comment (Camilo-Camargo) [inline internal/service/client/client.go:171]: mejor cambiar payload.id 64 a 32 para no realizar int32(...)

File: internal/service/mailing/templates/template-password.html
Comment (Camilo-Camargo) [inline internal/service/mailing/templates/template-password.html:1]: eliminar este archivo o agregarlo a gitignore

File: internal/service/mailing/templates/template-reset-password.mjml
Comment (Camilo-Camargo) [inline internal/service/mailing/templates/template-reset-password.mjml:1]: idem

File: internal/service/client/client.go [hunk 3]
```diff
@@ -115,21 +117,82 @@ func (s *serviceClient) Login(req *dto.LoginClientReq) (*dto.LoginClientRes, err
...
+}
+
+func (s *serviceClient) ResetPassword(req dto.ResetPasswordReq) (bool, error) {
+	res, err := s.clientRepository.LoginClient(req.Email)
+	if err != nil {
+		return false, err
+	}
+	client := &dto.ClientTokenResetPassword{
+		ID:  res.ClientID,
+		Exp: time.Now().Add(time.Hour / 2).Unix(),
```
[the remark this answers is not in the document]
Comment (CamiloArias12) [reply internal/service/client/client.go:147]: image
si

File: internal/service/client/client.go (hunk 2 above)
Comment (CamiloArias12) [reply internal/service/client/client.go]: [b7f059d](https://github.com/OwlByTech/optitech-api/pull/15/commits/b7f059dec3cf0ff0b400c88dc2d1d3242e105d59)

File: internal/service/mailing/templates/template-password.html
Comment (CamiloArias12) [reply internal/service/mailing/templates/template-password.html:1]: [0cc6cda](https://github.com/OwlByTech/optitech-api/pull/15/commits/0cc6cda2b724306d753742e7400a309e97b1752d)

File: internal/service/mailing/templates/template-reset-password.mjml
Comment (Camilo-Camargo) [reply internal/service/mailing/templates/template-reset-password.mjml:1]: [0cc6cda](https://github.com/OwlByTech/optitech-api/pull/15/commits/0cc6cda2b724306d753742e7400a309e97b1752d)

File: internal/service/mailing/templates/template-reset-password.mjml [hunk 4]
```diff
@@ -0,0 +1,27 @@
+<mjml>
+  <mj-body background-color="#f0f0f0">
+    <mj-section>
+      <mj-column>
+        <mj-text font-size="40px"> Opti Tech </mj-text>
+        <mj-divider border-color="#000000" border-width="2px" width="100%" />
+        <mj-text
+          >Bienvenido a Opti Tech, restablecer contraseña.</mj-text
```
Comment (JosephSC0121) [inline internal/service/mailing/templates/template-reset-password.mjml:8]:
```suggestion
          >Bienvenido a OptiTech, restablecer contraseña.</mj-text
```

**Review by JosephSC0121 — APPROVED** [review]

File: internal/security/jwt.go [hunk 5]
```diff
@@ -32,3 +33,16 @@ func JWTVerify(token string, secret string) (*jwt.Token, error) {
...
+	tokenParsed, err := jwt.ParseWithClaims(token, &payload, func(token *jwt.Token) (interface{}, error) {
+		return []byte(secret), nil
+	})
+
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return tokenParsed, &payload, nil
+}
```
[the remark this answers is not in the document]
Comment (Camilo-Camargo) [reply internal/security/jwt.go]: https://github.com/OwlByTech/optitech-api/pull/19

**Review by Camilo-Camargo — APPROVED** [review]

**Pull request merged**